governance/xc_admin/packages/xc_admin_common/src/programs/core/core_adapter.ts

```diff
@@ -0,0 +1,787 @@
+            address: product.address.toBase58(),
+            metadata: {
+              ...product.metadata,
+            },
+            priceAccounts: product.priceAccounts.map((p: PriceRawConfig) => {
+              return {
+                address: p.address.toBase58(),
+                publishers: p.publishers.map((p) => p.toBase58()),
+                expo: p.expo,
+                minPub: p.minPub,
+                maxLatency: p.maxLatency,
+              };
+            }),
+          };
+          // This field is immutable and should not be updated
+          delete symbolToData[product.metadata.symbol].metadata.price_account;
+        });
+    }
+
+    return this.sortData(symbolToData);
+  }
+
+  /**
+   * Sort configuration data for consistent output
+   */
+  private sortData(data: any) {
+    const sortedData: any = {};
+    Object.keys(data)
+      .sort()
+      .forEach((key) => {
+        const sortedInnerData: any = {};
+        Object.keys(data[key])
+          .sort()
+          .forEach((innerKey) => {
+            if (innerKey === "metadata") {
+              sortedInnerData[innerKey] = this.sortObjectByKeys(
+                data[key][innerKey],
+              );
+            } else if (innerKey === "priceAccounts") {
+              // Sort price accounts by address
+              sortedInnerData[innerKey] = data[key][innerKey].sort(
+                (priceAccount1: any, priceAccount2: any) =>
+                  priceAccount1.address.localeCompare(priceAccount2.address),
+              );
+              // Sort price accounts keys
+              sortedInnerData[innerKey] = sortedInnerData[innerKey].map(
+                (priceAccount: any) => {
+                  const sortedPriceAccount: any = {};
+                  Object.keys(priceAccount)
+                    .sort()
+                    .forEach((priceAccountKey) => {
+                      if (priceAccountKey === "publishers") {
+                        sortedPriceAccount[priceAccountKey] = priceAccount[
+                          priceAccountKey
+                        ].sort((pub1: string, pub2: string) =>
+                          pub1.localeCompare(pub2),
+                        );
+                      } else {
+                        sortedPriceAccount[priceAccountKey] =
+                          priceAccount[priceAccountKey];
+                      }
+                    });
+                  return sortedPriceAccount;
+                },
+              );
+            } else {
+              sortedInnerData[innerKey] = data[key][innerKey];
+            }
+          });
+        sortedData[key] = sortedInnerData;
+      });
+    return sortedData;
+  }
+
+  /**
+   * Sort object by keys
+   */
+  private sortObjectByKeys(obj: any) {
+    const sortedObj: any = {};
+    Object.keys(obj)
+      .sort()
+      .forEach((key) => {
+        sortedObj[key] = obj[key];
+      });
+    return sortedObj;
+  }
```

Comment:
I have 2 thoughts here:

1. [Javascript object property ordering is complex](https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order/38218582#38218582), engine dependent, brittle, and is not intended to be used to drive real code.  If you need guaranteed ordering, you should be using an array of tuples, not an object.

2. If you were to write this code, in getting into the habit of writing immutable, functional code, this kind of thing is better done using `Object.fromEntries` and `entries`, e.g. something like this:

```tsx
const sortObjectByKeys = (obj: Record<string, unknown>) =>
  Object.fromEntries(obj.entries().toSorted((a, b) => a[0].localeCompare(b[0])));
```